src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ILanguageServiceReferenceFinder.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.FindSymbols.Finders
+{
+    internal interface ILanguageServiceReferenceFinder : ILanguageService
+    {
+        Task<IEnumerable<ISymbol>> DetermineCascadedSymbolsAsync(INamedTypeSymbol symbol, Project project, CancellationToken cancellationToken);
```

Comment:
Why is this specific to INamedTypeSymbol? Don't you need to cascade from the property to the named type as well so it's symmetric?